packages/router/src/components/prompt/prompt.tsx

```diff
@@ -30,20 +37,21 @@ export class StencilRouterPrompt implements ComponentInterface {
   }
 
   componentWillLoad() {
-    if (this.when ) {
+    if (this.when) {
       this.enable(this.message);
     }
   }
 
-  @Watch('message')
-  @Watch('when')
-  updateMessage(newMessage: string, prevMessage: string) {
-    if (this.when) {
-      if (!this.when || prevMessage !== newMessage) {
-        this.enable(this.message);
-      }
-    } else {
+  updateMessage(newMessage: string, prevMessage: string): void; // `message` updated
+  updateMessage(newMessage: boolean, prevMessage: boolean): void; // `when` updated
+
+  @Watch("message")
+  @Watch("when")
+  updateMessage<T extends string | boolean>(newMessage: T, prevMessage: T) {
+    if (!this.when) {
       this.disable();
+    } else if (prevMessage !== newMessage) {
+      this.enable(this.message);
```

Comment:
Tried to simplify the logic here a little while being more earnest about the parameter types.